packages/flutter_tools/lib/src/commands/build_apk.dart

```diff
@@ -227,51 +224,43 @@ class BuildApkCommand extends FlutterCommand {
 
 Future<_ApkComponents> _findApkComponents(
   TargetPlatform platform,
-  BuildConfiguration config,
-  String enginePath,
+  BuildVariant buildVariant,
   String manifest,
-  String resources,
-  BuildVariant buildVariant
+  String resources
 ) async {
-  // TODO(devoncarew): Get the right artifacts for [buildVariant].
+  _ApkComponents components = new _ApkComponents();
+  components.manifest = new File(manifest);
+  components.resources = resources == null ? null : new Directory(resources);
 
-  List<String> artifactPaths;
-  if (enginePath != null) {
+  if (tools.isLocalEngine) {
     String abiDir = platform == TargetPlatform.android_arm ? 'armeabi-v7a' : 'x86_64';
-    artifactPaths = [
-      '$enginePath/third_party/icu/android/icudtl.dat',
-      '${config.buildDir}/gen/sky/shell/shell/classes.dex.jar',
-      '${config.buildDir}/gen/sky/shell/shell/shell/libs/$abiDir/libsky_shell.so',
-      '$enginePath/build/android/ant/chromium-debug.keystore',
+    String enginePath = tools.engineSrcPath;
+    String buildDir = tools.getEngineArtifactsDirectory(platform, buildVariant).path;
+
+    components.icuData = new File('$enginePath/third_party/icu/android/icudtl.dat');
+    components.jars = <File>[
+      new File('$buildDir/gen/sky/shell/shell/classes.dex.jar')
     ];
+    components.libSkyShell = new File('$buildDir/gen/sky/shell/shell/shell/libs/$abiDir/libsky_shell.so');
+    components.debugKeystore = new File('$enginePath/build/android/ant/chromium-debug.keystore');
   } else {
-    List<ArtifactType> artifactTypes = <ArtifactType>[
-      ArtifactType.androidIcuData,
-      ArtifactType.androidClassesJar,
-      ArtifactType.androidLibSkyShell,
-      ArtifactType.androidKeystore,
+    Directory artifacts = tools.getEngineArtifactsDirectory(platform, buildVariant);
+
+    components.icuData = new File(path.join(artifacts.path, 'icudtl.dat'));
+    components.jars = <File>[
+      new File(path.join(artifacts.path, 'classes.dex.jar'))
     ];
-    artifactPaths = artifactTypes.map((ArtifactType type) {
-      return ArtifactStore.getPath(ArtifactStore.getArtifact(
-        type: type,
-        targetPlatform: config.targetPlatform
-      ));
-    }).toList();
+    components.libSkyShell = new File(path.join(artifacts.path, 'libsky_shell.so'));
```

Comment:
I don't understand why some paths use the deep .so vs. this one.  I suspect one is stripped and one is not, and that one of these paths may be wrong.  @chinmaygarde or @jason-simmons might know.